adapters/oidc/adapter-core/src/test/java/org/keycloak/adapters/KeycloakDeploymentTest.java

```diff
@@ -68,6 +75,64 @@ public void stripDefaultPorts() {
         assertEquals("https://localhost/auth", keycloakDeployment.getAuthServerBaseUrl());
     }
 
+    @Test
+    public void logoutUrlNotNullInMultiThreading() throws InterruptedException {
+        KeycloakDeployment keycloakDeployment = new KeycloakDeploymentMock();
+        keycloakDeployment.setRealm("test");
+        AdapterConfig config = new AdapterConfig();
+        config.setAuthServerUrl("http://localhost:80/auth");
+        keycloakDeployment.setAuthServerBaseUrl(config);
+
+        int numberOfThreads = 1000;
```

Comment:
Maybe too many, don't you think?
I would prefer to use something like 64 (or similar) but repeat the test more times (another 64 or more if needed) instead of just once. Check if with those settings you can reproduce the issue without the fix. If you see the test doesn't fail start increasing threads and/or executions times.

Reply:
Fixed, at my environment 20 threads are enough to reproduce the original issue (usually when attempts <10), so I implemented 15 attempts of 20 threads which always caught the problem with the original code (at least in my PC).